pkg/autoscaler/autoscaler.go

```diff
@@ -457,6 +461,65 @@ var _ = Describe("Autoscaler should", framework.LabelAutoscaler, Serial, func()
 				cleanupObjects[asr.GetName()] = asr
 			}
 
+			for _, machineSet := range transientMachineSets {
+				By(fmt.Sprintf("Waiting for MachineSet %s to acquire autoscaling annotations", machineSet.GetName()))
+				Eventually(komega.Object(machineSet), framework.WaitShort, pollingInterval).Should(HaveField("GetAnnotations()", SatisfyAll(
+					HaveKey(caMinSizeAnnotation),
+					HaveKey(caMaxSizeAnnotation),
+				)), "MachineSet %s did not acquire autoscaling annotation", machineSet.GetName())
+			}
+
+			// on some cloud providers, we have experienced the new nodes having resources added to their status.capacity
+			// after they have been initialized. this has been seen with `hugepages-1Gi` and `hugepages-2Mi`. so we
+			// wait until the nodes have both entries before moving on.
+			By("Waiting for the new Nodes to have hugepages-1Gi and hugepages-2Mi capacity")
+			Eventually(func() ([]*corev1.Node, error) {
+				nodes := []*corev1.Node{}
+
+				for _, machineSet := range transientMachineSets {
+					if n, err := framework.GetNodesFromMachineSet(client, machineSet); err != nil {
+						return nodes, err
+					} else if len(n) != 1 {
+						return nodes, fmt.Errorf("expected 1 node for MachineSet %s, found %d", machineSet.GetName(), len(n))
+					} else {
+						nodes = append(nodes, n[0])
+					}
+				}
+
+				return nodes, nil
+			}, framework.WaitMedium, pollingInterval).Should(HaveEach(
+				HaveField("Status.Capacity", SatisfyAny(
+					HaveKey(corev1.ResourceHugePagesPrefix+"1Gi"),
+					HaveKey(corev1.ResourceHugePagesPrefix+"2Mi"),
+				)),
```

Comment:
@JoelSpeed did i mess up this logic?

it seems like the test is failing but the output contains the expected fields.

Reply:
The issue is typing, the keys you've put in are `string` where the keys in the map are `corev1.ResourceName`, which, happens to be `type ResourceName string`, so, a `BeEquivalentTo` will solve your issue, as this will cast the key before it compares

Reply:
wow, good call